### PR TITLE
fix: pass cloud user id and cloud instance id while creating metrics client

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description:
Earlier we would pass it per event (only one event supported) - now we get it for every event

Part of #1619

### CODE REVIEW

File: api/golang/core/lib/starlark_run_config/starlark_run_config.go [hunk 1]
```diff
@@ -85,12 +85,14 @@ func WithExperimentalFeatureFlags(experimentalFeatureFlags []kurtosis_core_rpc_a
 	}
 }
 
+// Deprecated: This isn't used anymore and the value passed will not be read anywhere
 func WithCloudInstanceId(cloudInstanceId string) starlarkRunConfigOption {
```
Comment (leoporoli) [inline api/golang/core/lib/starlark_run_config/starlark_run_config.go:89]: are you tracking the remotion in a ticket?

File: core/launcher/args/api_container_args.go [hunk 2]
```diff
@@ -47,8 +48,16 @@ type APIContainerArgs struct {
...
 	IsCI bool `json:"is_ci"`
+
+	// The Cloud User ID of the current user if available
+	CloudUserID metrics_client.CloudUserID `json:"cloud_user_id"`
+
+	// The Cloud Instance ID of the current user if available
+	CloudInstanceID metrics_client.CloudInstanceID `json:"cloud_instance_id"`
 }
 
+var skipValidation = []string{"cloud_user_id", "cloud_instance_id"}
```
Comment (leoporoli) [inline core/launcher/args/api_container_args.go]: nit: you could use a set (map[string]bool) instead and avoid having to iterate the slice

File: engine/launcher/args/args.go [hunk 3]
```diff
@@ -50,8 +51,16 @@ type EngineServerArgs struct {
...
 	IsCI bool `json:"is_ci"`
+
+	// The Cloud User ID of the current user if available
+	CloudUserID metrics_client.CloudUserID `json:"cloud_user_id"`
+
+	// The Cloud Instance ID of the current user if available
+	CloudInstanceID metrics_client.CloudInstanceID `json:"cloud_instance_id"`
 }
 
+var skipValidation = []string{"cloud_user_id", "cloud_instance_id"}
```
Comment (leoporoli) [inline engine/launcher/args/args.go]: same here, you could use a set instead

File: engine/launcher/args/args.go [hunk 4]
```diff
@@ -136,3 +153,12 @@ func (args EngineServerArgs) validate() error {
 	}
 	return nil
 }
+
+func contains(slice []string, value string) bool {
```
Comment (leoporoli) [inline engine/launcher/args/args.go]: you can remove this helper function if you use a set instead

File: metrics-library/golang/lib/metrics_client/metrics_client_types.go [hunk 5]
```diff
@@ -7,3 +7,6 @@ const (
 )
 
 type MetricsClientType string
+
+type CloudUserID string
```
Comment (leoporoli) [inline metrics-library/golang/lib/metrics_client/metrics_client_types.go:11]: nice!

**Review by leoporoli — APPROVED** [review]

File: engine/launcher/args/args.go (hunk 4 above)
Comment (h4ck3rk3y) [reply engine/launcher/args/args.go]: thanks! this makes sense :)

**Pull request merged**